grapple/types/redirects.py

```diff
@@ -1,31 +1,52 @@
+import json
+
+from typing import Optional
+
 import graphene
 
-from django.conf import settings
 from wagtail.contrib.redirects.models import Redirect
+from wagtail.models import Page, Site
 
 from .pages import get_page_interface
 
 
 class RedirectType(graphene.ObjectType):
     old_path = graphene.String(required=True)
     old_url = graphene.String(required=True)
-    new_url = graphene.String(required=True)
+    new_url = graphene.String(required=False)
     page = graphene.Field(get_page_interface())
     is_permanent = graphene.Boolean(required=True)
 
-    # Give old_path with BASE_URL attached.
-    def resolve_old_url(self, info, **kwargs):
-        return settings.BASE_URL + self.old_path
+    def resolve_old_url(self, info, **kwargs) -> str:
+        """
+        If the redirect is for a specific site, append hostname and port to path.
+        Otherwise, return a JSON string representing a list of all site urls.
+        """
+        if self.site:
+            return f"http://{self.site.hostname}:{self.site.port}/{self.old_path}"
+
+        if self.site is None:
+            sites_QS = Site.objects.all()
```

Comment:
**Issue:** Editors control what sites a redirect applies to from within Wagtail. So, every redirect created within Wagtail will either be assigned to a single site or "all sites" (which is the empty/none value). 

However, this code ignores that and just fetches all `Site`s in the database, negating editors' choice.

Reply:
My understanding is that not specifying `site` means that a redirect is assigned to "all sites". In this case, we are getting all sites, to return a list of sites the redirect applies to.